c_reference/tests/kws/test_keyword_spotting.c

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
+#include<time.h>
+#include"conv1d.h"
+#include"dscnn.h"
+#include"fastgrnn.h"
+#include"utils.h"
+
+#include"keyword_spotting_io_2.h"
+#include"precnn_params.h"
+#include"rnn_params.h"
+#include"postcnn_params.h"
+
+// Check out time steps with label time steps
+int checkTime(unsigned out_T) {
+  if (out_T != O_T) {
+    printf("Error, Estimated Output and Actual ouput time axis mis-match");
+    return 1;
+  }
+  return 0;
+}
+// Error Check
+void checkError(float* pred, float* label) {
+  float error = 0, denom = 0;
+  for (int t = 0; t < O_T; t++) {
+    for (int d = 0; d < POST_CNN_O_F; d++) {
+      error += ((pred[t * POST_CNN_O_F + d]-label[t * POST_CNN_O_F + d])*(pred[t * POST_CNN_O_F + d]-label[t * POST_CNN_O_F + d]));
+      denom += label[t * POST_CNN_O_F + d] * label[t * POST_CNN_O_F + d];
+    }
+  }
+  float avg_error = error/(O_T*POST_CNN_O_F);
+  printf("Full Network\n");
+  printf("Aggregate Squared Error : %f   ;   Mean Sqaured Error : %f  \n", error, avg_error);
+  printf("Denominator for RMSE : %f \n", denom);
```

Comment:
What is the significance of the denom and why must the user know it?

Reply:
Sorry, was using it for debugging. I've removed the line